### PR TITLE
xbox: Remove XVideoSetMode from XboxCRTEntry

### PR DESCRIPTION
**Pull request by dracc**
This should be done per application, not before the application has launched.

### CODE REVIEW

File: platform/xbox/crt0.c [hunk 1]
```diff
@@ -19,8 +19,6 @@ static int main_wrapper ()
 
 void XboxCRTEntry ()
 {
-    XVideoSetMode(640, 480, 32, REFRESH_DEFAULT);
```
Comment (JayFoxRox) [inline platform/xbox/crt0.c:52]: Should be able to also remove `#include <hal/video.h>` now

Comment (dracc) [reply platform/xbox/crt0.c:52]: Done!

**Comment by dracc** [comment]
I believe most (if not all) major stake holders have been prepared for this change. Let's pull the plug soon:tm:

**Comment by JayFoxRox** [comment]
Most of them didn't merge yet.

I believe they could merge regardless of this change - the only thing that happens is that the existing mode will be set twice (in quick succession).

This should be tested and then this can marinade for a while.

*This change was also primarily motivated by myself, and other maintainers might have other / better ideas (although I consider this change unproblematic); they should get a chance to look at this - which might take a while.*

**Comment by JayFoxRox** [comment]
We'll wait with this one until #4 has completed; we want 2 separate submodule updates in nxdk (1. pdclib update / 2. XVideoSetMode removal).

**Comment by JayFoxRox** [comment]
**Important note for the follow-up nxdk PR (when updating submodule):**

[`vmCurrent` in lib/hal/video.c](https://github.com/XboxDev/nxdk/blob/master/lib/hal/video.c#L35) will be uninitialized until the first call of `XVideoSetMode` (the same happens if `XVideoSetMode` couldn't find a mode).

A temporary solution will be to initialize it to an unknown state, and to disallow certain APIs (including `debugPrint`) until `XVideoSetMode` was explicitly called.

A possible permanent solution would be to refactor all APIs, or to detect if the mode is unknown, and read the hardware state.

[We should be able to recover the `pVidMode->dwMode` from XVideoInit](https://github.com/XboxDev/nxdk/blob/master/lib/hal/video.c#L332), using:

```c
ULONG mode;
AvSendTVEncoderOption(0xFD000000, 13, 0, &mode);
```

That should give us the width and height.

For the framebuffer format (bits per pixel) we'll probably have to query the `CR[0x28]` and `RAMDAC` registers.

**Comment by thrimbor** [comment]
I think enough time has passed to go ahead with this. Merging.

**Pull request merged**